### DISCUSSION TITLE
How to keep a connection alive?

### DISCUSSION THREAD

**Discussion by akparhi** [prisma/prisma#4653] [Q&A] [answered]
Prisma does not seem to have an option to keep a connection alive. I've tried using `prisma.$connect()` and still first request takes quite longer to resolve. Also roughly after 5 seconds connection seems to be idle and this results in 10x slower queries if there has been no queries in last 5 seconds.

![image](https://user-images.githubusercontent.com/22069968/102286372-2aa71200-3f5e-11eb-978b-34f60b947ac3.png)

**Comment by ryands17** [comment]
Hey @logicaltitan 👋 

Are you using the latest version of Prisma Client and CLI (2.13.0)? If not, then do update to the latest one as `connect` should automatically create a connection to the database.

Th reason is that this behaviour was broken in the previous release and has been fixed in the latest one. Here is the related [issue](https://github.com/prisma/prisma/issues/4068) and [release notes](https://github.com/prisma/prisma/releases/tag/2.13.0) that mentions the same under __Fixes and improvements__.

**Reply by akparhi** [reply]
Hey @ryands17 
Yep I'm using 2.13.0(Installed yesterday) and still have this issue. I'm not very sure about the behaviour of `prisma.$connect()`. Is it supposed to keep one connection always open like I expect it to?

**Reply by akparhi** [reply]
![image](https://user-images.githubusercontent.com/22069968/102344622-30ceda00-3fc2-11eb-8933-533eefe00528.png)

Steps to reproduce: 
1. clone `https://github.com/logicaltitan/filefy-be-exp`
2. Install deps with `yarn` and start server with `yarn dev`
3. GET request to `localhost:5050/auth/list-users` repeatedly with gaps of few seconds.

In this boilerplate I was using sequelize until yesterday and had no issues keeping an open connection. 
I realize lazy connection is nice if you have many concurrent users but this is a huge issue for low user apps, since every request with 5-6 second gaps can take 1.3s or more.

**Reply by akparhi** [reply]
Alright I did a bit more digging and it seems like this issue is not happening with postgresql. Since I'm at the early stage of building the app and I like prisma migrations a lot I'll just go with postgresql. However the above issue probably needs its own issue.

**Reply by ryands17** [reply, marked as answer]
@logicaltitan In that case, I would suggest creating an issue for this [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) with the reproduction steps. Also it would be great if you could mention a note there about your finding that this doesn't happen with Postgres. We would like to investigate this issue and come up with a solution for this :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
